src/core/Akka.Cluster/Cluster.cs

```diff
@@ -264,27 +267,24 @@ public void Join(Address address)
         /// <returns>Task which completes, once current cluster node reaches <see cref="MemberStatus.Up"/> state.</returns>
         public Task JoinAsync(Address address, CancellationToken token = default)
         {
-            var completion = new TaskCompletionSource<NotUsed>(TaskCreationOptions.RunContinuationsAsynchronously);
-            
-            var timeout = Settings.RetryUnsuccessfulJoinAfter ?? TimeSpan.FromSeconds(10);
-            var timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(token);
-            timeoutCts.CancelAfter(timeout);
-            timeoutCts.Token.Register(() =>
+            if (_asyncJoinState != null)
             {
-                timeoutCts.Dispose();
-                completion.TrySetException(new ClusterJoinFailedException(
-                    $"Node has not managed to join the cluster using provided address: {address}"));
-            });
+                _log.Error("Another async cluster join is already in progress");
```

Comment:
I'd just log a `DEBUG` here and return the task without making a fuss, in case this came up.